pkg/cluster/internal/kubeadm/config.go

```diff
@@ -94,8 +95,8 @@ type DerivedConfigData struct {
 	AdvertiseAddress string
 	// DockerStableTag is automatically derived from KubernetesVersion
 	DockerStableTag string
-	// SortedFeatureGateKeys allows us to iterate FeatureGates deterministically
-	SortedFeatureGateKeys []string
+	// SortedFeatureGates allows us to iterate FeatureGates deterministically
+	SortedFeatureGates []FeatureGate
```

Comment:
minor rework, because index is not supported. We didn't need it much other than cross-indexing featuregates by the sorted keys. So instead we just store sorted key + value.

Reply:
we could consider sorting both map forms, but currently we don't, no change there.